modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectDefinitionSettingUtil.java

```diff
@@ -35,41 +38,89 @@ public class ObjectDefinitionSettingUtil {
+						serviceBuilderObjectDefinitionSetting,
+						objectDefinitionSetting);
+				}
+
+				if (StringUtil.equals(
 						ObjectDefinitionSettingConstants.
 							NAME_ACCEPTED_GROUP_EXTERNAL_REFERENCE_CODES,
 						objectDefinitionSetting.getName())) {
 
 					serviceBuilderObjectDefinitionSetting.setName(
-						objectDefinitionSetting.getName());
+						ObjectDefinitionSettingConstants.
+							NAME_ACCEPTED_GROUP_IDS);
+
+					String groupExternalReferenceCodes = String.valueOf(
+						objectDefinitionSetting.getValue());
+
 					serviceBuilderObjectDefinitionSetting.setValue(
-						String.valueOf(objectDefinitionSetting.getValue()));
+						StringUtil.merge(
+							TransformUtil.transform(
+								groupExternalReferenceCodes.split("\\s*,\\s*"),
+								groupERC -> {
+									Group group =
+										groupLocalService.
+											getGroupByExternalReferenceCode(
+												groupERC, companyId);
+
+									return String.valueOf(group.getGroupId());
+								},
+								String.class)));
 
 					return serviceBuilderObjectDefinitionSetting;
 				}
 
 				serviceBuilderObjectDefinitionSetting.setName(
-					ObjectDefinitionSettingConstants.NAME_ACCEPTED_GROUP_IDS);
+					ObjectDefinitionSettingConstants.
+						NAME_ROOT_OBJECT_DEFINITION_IDS);
 
-				String groupExternalReferenceCodes = String.valueOf(
-					objectDefinitionSetting.getValue());
+				String rootObjectDefinitionExternalReferenceCodes =
+					String.valueOf(objectDefinitionSetting.getValue());
 
 				serviceBuilderObjectDefinitionSetting.setValue(
 					StringUtil.merge(
 						TransformUtil.transform(
-							groupExternalReferenceCodes.split("\\s*,\\s*"),
-							groupERC -> {
-								Group group =
-									groupLocalService.
-										getGroupByExternalReferenceCode(
-											groupERC, companyId);
-
-								return String.valueOf(group.getGroupId());
+							rootObjectDefinitionExternalReferenceCodes.split(
+								"\\s*,\\s*"),
+							rootObjectDefinitionExternalReferenceCode -> {
+								ObjectDefinition objectDefinition =
+									objectDefinitionLocalService.
+										getObjectDefinitionByExternalReferenceCode(
+											rootObjectDefinitionExternalReferenceCodes,
+											companyId);
+
+								return String.valueOf(
+									objectDefinition.getObjectDefinitionId());
 							},
 							String.class)));
 
 				return serviceBuilderObjectDefinitionSetting;
 			});
 	}
 
+	private static com.liferay.object.model.ObjectDefinitionSetting
+		_setObjectDefinitionsSettingsByExternalReferenceCode(
+			com.liferay.object.model.ObjectDefinitionSetting
+				serviceBuilderObjectDefinitionSetting,
+			ObjectDefinitionSetting objectDefinitionSetting) {
+
+		serviceBuilderObjectDefinitionSetting.setName(
+			objectDefinitionSetting.getName());
+		serviceBuilderObjectDefinitionSetting.setValue(
+			String.valueOf(objectDefinitionSetting.getValue()));
+
+		return serviceBuilderObjectDefinitionSetting;
+	}
```

Comment:
@victorKammerer since we only use this method in one place and it is not that big perform its logic in the place it is called.

Thanks!